packages/run-protocol/src/vpool-xyk-amm/addPool.js

```diff
@@ -29,33 +37,67 @@ export const makeAddIssuer = (
       E(secondaryIssuer).getAssetKind(),
       E(secondaryIssuer).getBrand(),
     ]);
+    // AWAIT ///////////////
 
-    assert(
-      !isInSecondaries(secondaryBrand),
-      X`issuer ${secondaryIssuer} already has a pool`,
-    );
     assert(
       secondaryAssetKind === AssetKind.NAT,
       X`${keyword} asset not fungible (must use NAT math)`,
     );
+
+    if (brandToLiquidityIssuer.has(secondaryBrand)) {
+      return brandToLiquidityIssuer.get(secondaryBrand);
+    }
+
+    assert(
+      !isInSecondaries(secondaryBrand),
+      X`issuer ${secondaryIssuer} already has a pool`,
+    );
+
+    if (brandToLiquidityMint.has(secondaryBrand)) {
+      const { issuer } = brandToLiquidityMint
+        .get(secondaryBrand)
+        .getIssuerRecord();
+      return issuer;
+    }
+
+    /** @template T @typedef {import('@endo/promise-kit').PromiseKit<T>} PromiseKit */
+    /** @type {PromiseKit<Issuer<'nat'>>} */
```

Comment:
I think this can be simplified to:
```suggestion
    /** @type {import('@endo/promise-kit').PromiseKit<Issuer<'nat'>>} */
```

Reply:
done